zerver/migrations/0584_namedusergroup_creator_date_created_backfill.py

```diff
@@ -12,30 +13,26 @@ def backfill_creator_id_and_date_created_from_realm_audit_log(
     RealmAuditLog.USER_GROUP_CREATED = 701
     NamedUserGroup = apps.get_model("zerver", "NamedUserGroup")
 
-    user_group_creator_updates = []
-    for audit_log_entry in RealmAuditLog.objects.select_related("modified_user_group").filter(
-        event_type=RealmAuditLog.USER_GROUP_CREATED,
-        acting_user_id__isnull=False,
-    ):
-        assert audit_log_entry.modified_user_group is not None
-        user_group = audit_log_entry.modified_user_group
-        user_group.creator_id = audit_log_entry.acting_user_id
-        user_group_creator_updates.append(user_group)
-
-    NamedUserGroup.objects.bulk_update(user_group_creator_updates, ["creator_id"], batch_size=1000)
-
-    user_group_date_created_updates = []
-    for audit_log_entry in RealmAuditLog.objects.select_related("modified_user_group").filter(
-        event_type=RealmAuditLog.USER_GROUP_CREATED,
-        event_time__isnull=False,
-    ):
-        assert audit_log_entry.modified_user_group is not None
-        user_group = audit_log_entry.modified_user_group
-        user_group.date_created = audit_log_entry.event_time
-        user_group_date_created_updates.append(user_group)
-
-    NamedUserGroup.objects.bulk_update(
-        user_group_date_created_updates, ["date_created"], batch_size=1000
+    NamedUserGroup.objects.annotate(
+        new_creator=Subquery(
+            RealmAuditLog.objects.filter(
+                event_type=RealmAuditLog.USER_GROUP_CREATED,
+                acting_user__isnull=False,
+                modified_user_group=OuterRef("pk"),
+            ).values("acting_user")
+        )
+    ).filter(creator=None, new_creator__isnull=False).update(creator=F("new_creator"))
+
+    NamedUserGroup.objects.annotate(
+        new_date_created=Subquery(
+            RealmAuditLog.objects.filter(
+                event_type=RealmAuditLog.USER_GROUP_CREATED,
+                event_time__isnull=False,
+                modified_user_group=OuterRef("pk"),
+            ).values("event_time")
+        )
+    ).filter(date_created=None, new_date_created__isnull=False).update(
+        date_created=F("new_date_created")
     )
```

Comment:
Is there a reason we separated the `creator` loop from the `date_created` loop (i.e., do we expect these to be backfilled in distinct sets of `NamedUserGroup`s)?

Reply:
I think 63f593620740343cb2b51c0e1c1336b60816cc45 suggests some groups having `acting_user=None` even though they have an event time, so they may not be identical sets.